src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java

```diff
@@ -0,0 +1,112 @@
+package gov.cabinetofice.gapuserservice.service;
+
+import gov.cabinetofice.gapuserservice.exceptions.AuthenticationException;
+import gov.cabinetofice.gapuserservice.exceptions.InvalidRequestException;
+import gov.cabinetofice.gapuserservice.exceptions.PrivateKeyParsingException;
+import gov.cabinetofice.gapuserservice.util.RestUtils;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import lombok.RequiredArgsConstructor;
+import org.apache.http.HttpHeaders;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class OneLoginService {
+
+    @Value("${onelogin.client-id}")
+    private String clientId;
+
+    @Value("${onelogin.base-url}")
+    private String oneLoginBaseUrl;
+
+    @Value("${onelogin.client-assertion-type}")
+    private String clientAssertionType;
+
+    @Value("${onelogin.service-redirect-url}")
+    private String serviceRedirectUrl;
+
+    @Value("${onelogin.private-key}")
+    public String privateKey;
```

Comment:
Could pull these in from a config bean instead, but pretty indifferent on this

Reply:
I feel like this way is more explicit and readable